plugins/sql.c

```diff
@@ -472,6 +482,71 @@ static struct command_result *json_sql(struct command *cmd,
 	return refresh_tables(cmd, dbq);
 }
 
+static struct command_result *json_sql(struct command *cmd,
+				       const char *buffer,
+				       const jsmntok_t *params)
+{
+	struct db_query *dbq = tal(cmd, struct db_query);
+	const char *query;
+
+	if (!param(cmd, buffer, params,
+		   p_req("query", param_string, &query),
+		   NULL))
+		return command_param_failed();
+
+	dbq->colnames = NULL;
+
+	return do_query(cmd, dbq, query);
+}
+
+static struct command_result *param_string_arr(struct command *cmd,
+					       const char *name,
+					       const char *buffer,
+					       const jsmntok_t *tok,
+					       char ***strings)
+{
+	size_t i;
+	const jsmntok_t *t;
+
+	if (tok->type != JSMN_ARRAY)
+		return command_fail_badparam(cmd, name, buffer, tok,
+					     "should be an array");
+
+	*strings = tal_arr(cmd, char *, tok->size);
+	json_for_each_arr(i, t, tok) {
+		if (t->type != JSMN_STRING)
+			return command_fail_badparam(cmd, name, buffer, t,
+						     "should be a string");
+		(*strings)[i] = json_strdup(*strings, buffer, t);
+	}
+	return NULL;
+}
+
+static struct command_result *json_sqlsimple(struct command *cmd,
+					     const char *buffer,
+					     const jsmntok_t *params)
+{
+	struct db_query *dbq = tal(cmd, struct db_query);
+	const char *postquery;
+	char *query;
+
+	if (!param(cmd, buffer, params,
+		   p_req("columns", param_string_arr, &dbq->colnames),
+		   p_req("query", param_string, &postquery),
+		   NULL))
+		return command_param_failed();
+
+	query = tal_strdup(dbq, "SELECT ");
+	for (size_t i = 0; i < tal_count(dbq->colnames); i++) {
+		tal_append_fmt(&query, "%s%s ",
+			       i == 0 ? "" : ",",
+			       dbq->colnames[i]);
+	}
+	tal_append_fmt(&query, "%s", postquery);
```

Comment:
That's not the issue, since the authorize step happens after (and they can already try a query: "; attach lightningd.sqlite3; delete from channels" using the normal "sql" commands.  The only difference is that sqlsimple provides names for the columns.  Prepending it to the command is just a courtesy, really (we could require they do that themselves).